src/Microsoft.DotNet.Interactive.PowerShell.Tests/PowerShellKernelTests.cs

```diff
@@ -216,5 +217,27 @@ public async Task VerifyAllUsersProfileRuns()
                 File.Delete(_allUsersCurrentHostProfilePath);
             }
         }
+
+        [Fact]
+        public async Task Powershell_customobject_is_parsed_for_outdisplay()
+        {
+            var props = (new Dictionary<string, object>
+            {
+                { "prop1", "value1" },
+                { "prop2", "value2" },
+                { "prop3", "value3" }
+            });
+
+            var kernel = CreateKernel(Language.PowerShell);
+            var result = await kernel.SendAsync(new SubmitCode("[pscustomobject]@{ prop1 = 'value1'; prop2 = 'value2'; prop3 = 'value3' } | Out-Display"));
+            var outputs = result.KernelEvents.ToSubscribedList();
+
+            outputs.Should().SatisfyRespectively(
+                e => e.Should().BeOfType<CodeSubmissionReceived>(),
+                e => e.Should().BeOfType<CompleteCodeSubmissionReceived>(),
+                e => e.Should().BeOfType<DisplayedValueProduced>().Which.Value.Should().BeEquivalentTo(props),
```

Comment:
The `Value` property will no longer be serialized soon (#512), so it's better here to verify the `FormattedValues`, e.g.

```cs
 e => e.Should().BeOfType<DisplayedValueProduced>()
    .Which                      
    .FormattedValues
    .Should()
    .ContainSingle(f => f.Value == ... )
```